perfmetrics/scripts/testing_on_gke/examples/run-gke-tests.sh

```diff
@@ -0,0 +1,530 @@
+  printHelp
+  exitWithSuccess
+fi
+
+# Set environment variables.
+# GCP related
+test -n "${project_id}" || export project_id=${DEFAULT_PROJECT_ID}
+test -n "${project_number}" || export project_number=${DEFAULT_PROJECT_NUMBER}
+test -n "${zone}" || export zone=${DEFAULT_ZONE}
+# GKE cluster related
+test -n "${cluster_name}" || export cluster_name=${DEFAULT_CLUSTER_NAME}
+test -n "${node_pool}" || export node_pool=${DEFAULT_NODE_POOL}
+test -n "${machine_type}" || export machine_type=${DEFAULT_MACHINE_TYPE}
+test -n "${num_nodes}" || export num_nodes=${DEFAULT_NUM_NODES}
+test -n "${num_ssd}" || export num_ssd=${DEFAULT_NUM_SSD}
+test -n "${appnamespace}" || export appnamespace=${DEFAULT_APPNAMESPACE}
+test -n "${ksa}" || export ksa=${DEFAULT_KSA}
+test -n "${use_custom_csi_driver}" || export use_custom_csi_driver="${DEFAULT_USE_CUSTOM_CSI_DRIVER}"
+# GCSFuse/GKE GCSFuse CSI Driver source code related
+test -n "${src_dir}" || export src_dir=${DEFAULT_SRC_DIR}
+test -d "${src_dir}" || mkdir -pv "${src_dir}"
+test -n "${gcsfuse_src_dir}" || export gcsfuse_src_dir="${src_dir}"/gcsfuse
+export gke_testing_dir="${gcsfuse_src_dir}"/perfmetrics/scripts/testing_on_gke
+test -n "${csi_src_dir}" || export csi_src_dir="${src_dir}"/gcs-fuse-csi-driver
+# GCSFuse configuration related
+test -n "${gcsfuse_mount_options}" || export gcsfuse_mount_options="${DEFAULT_GCSFUSE_MOUNT_OPTIONS}"
+# Test runtime configuration
+test -n "${pod_wait_time_in_seconds}" || export pod_wait_time_in_seconds="${DEFAULT_POD_WAIT_TIME_IN_SECONDS}"
+
+function printRunParameters() {
+  echo "Running $0 with following parameters:"
+  echo ""
+  # GCP related
+  echo "project_id=\"${project_id}\""
+  echo "project_number=\"${project_number}\""
+  echo "zone=\"${zone}\""
+  # GKE cluster related
+  echo "cluster_name=\"${cluster_name}\""
+  echo "node_pool=\"${node_pool}\""
+  echo "machine_type=\"${machine_type}\""
+  echo "num_nodes=\"${num_nodes}\""
+  echo "num_ssd=\"${num_ssd}\""
+  echo "appnamespace=\"${appnamespace}\""
+  echo "ksa=\"${ksa}\""
+  echo "use_custom_csi_driver=\"${use_custom_csi_driver}\""
+  # GCSFuse/GKE GCSFuse CSI Driver source code related
+  echo "src_dir=\"${src_dir}\""
+  echo "gcsfuse_src_dir=\"${gcsfuse_src_dir}\""
+  echo "csi_src_dir=\"${csi_src_dir}\""
+  # GCSFuse configuration related
+  echo "gcsfuse_mount_options=\"${gcsfuse_mount_options}\""
+  echo "${gcsfuse_mount_options}" >gcsfuse_mount_options
+  # Test runtime configuration
+  echo "pod_wait_time_in_seconds=\"${pod_wait_time_in_seconds}\""
+  echo ""
+  echo ""
+  echo ""
+}
+
+# Install dependencies.
+function installDependencies() {
+  which helm || (cd "${src_dir}" && (test -d "./helm" || git clone https://github.com/helm/helm.git) && cd helm && make && ls -lh bin/ && mkdir -pv ~/bin && cp -fv bin/helm ~/bin/ && chmod +x ~/bin/helm && export PATH=$PATH:$HOME/bin && echo $PATH && which helm && cd - && cd -)
```

Comment:
Is `make` guaranteed to be installed or do we need to install it?

Reply:
No, make is not guaranteed to be installed. I had added the remaining dependencies in a future PR in this sequence. I have now advanced the installation of all the dependencies here.